Source/SequenceExtensions.cs

```diff
@@ -20,7 +23,28 @@ public static class SequenceExtensions
 			Expression<Func<TMock, TResult>> expression)
 			where TMock : class
```

Comment:
This new implementation of `SetupSequence` is really ugly at the moment and needs more work. Can this be cleaned up somehow? And does this (method calls and property getters) cover all cases? What about e.g. indexers? Do matchers work correctly if we install a setup like this? I don't actually understand just what is happening here. :-/